snow/validation/retriever.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (C) 2019-2021, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package validation
+
+import (
+	"fmt"
+
+	"github.com/flare-foundation/flare/ids"
+	lru "github.com/hashicorp/golang-lru"
+)
+
+const (
+	validatorSetsCacheSize = 64
+)
+
+type Retriever interface {
+	GetValidators(blockID ids.ID) (Set, error)
+}
+
+type cachingRetriever struct {
+	retriever Retriever
+	cache     *lru.Cache
+}
+
+func NewCachingRetriever(retriever Retriever) Retriever {
+
+	cache, _ := lru.New(validatorSetsCacheSize)
```

Comment:
Should the error be checked even though currently we hard code `validatorSetsCacheSize`?

Reply:
A cache with zero size is the only error condition, in which case it still works, but caches nothing. No need to check for an error here.